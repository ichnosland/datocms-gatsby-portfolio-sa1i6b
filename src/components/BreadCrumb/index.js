/**
*
* BreadCrumb
*
*/

import styled, { css } from 'styled-components';

import Div from 'components/Div';
import media from 'style/mediainjector';
import { colore } from 'style/color';

export const BreadCrumb = styled(Div)`
  color: ${(props) => props.color};
  position: relative;
  display: ${(props) => props.display ? props.display : 'inline-block'};
  text-align: center;
  border: ${(props) => props.border ? props.border : `1px solid ${colore.ui.neutralBg}`};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.general};
  ${media.print`
   ${({ noPrint }) => noPrint && css`
       display:none;
   `}
 `}
`;

export const Crumb = styled(Div)`
  position: relative;
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  align-content: center;
  flex-wrap: nowrap;
  padding: 0.4em 1.4em 0.4em 0.8em;
  &:first-child {
    padding-left: 1.2em;
    &:after {
      content: "⟩";
      position: absolute;
      right: 0;
    }
  }
  &:not(:first-child):not(:last-child) {
    &:after {
      content: "⟩";
      position: absolute;
      right: 0;
    }
  }
  &:only-child {
    &:after {
      content: "";
      position: relative;
    }
  }
`;

BreadCrumb.defaultProps = {
  theme: {
    radius: {
      general: '6px',
    },
  },
};

export default BreadCrumb;
