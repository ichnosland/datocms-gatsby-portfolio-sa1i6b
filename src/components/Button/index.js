/**
*
* Button
*
*/

import styled, { css } from 'styled-components';
//import { lighten, darken, transparentize } from 'polished';
import media from 'style/mediainjector';
import { colore, transparent } from 'style/color';
import Svg from '../Svg';

export const ButtonStyle = css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  appearance: none;
  font-family: ${(props) => props.theme.fonts.buttons.fontFamily};
  font-weight: ${(props) => props.theme.fonts.buttons.fontWeight};
  ${({ full }) => full && 'width: 100% !important;'};
  ${({ half }) => half && 'width: 50% !important;'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  text-align: center;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding ? props.padding : '0.4em  1.4em'};
  border: ${(props) => props.border ? props.border : '1px solid transparent'};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : props.theme.brand};
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.buttons};
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : props.theme.brand))};
    color: ${(props) => transparentize(0.5, (props.color ? props.color : colore.ui.contrast))};
    pointer-events: none;
    svg {
        opacity: .3;
      }
  }
  ${({ round }) => round && css`
    padding: 0;
    width: ${(props) => props.width};
    height: ${(props) => props.width};
    border-radius: 100%;
  `}
  ${({ actioncolor }) => actioncolor && css`
    background-color: ${colore.actions[actioncolor]};
    &:disabled {
      background-color: ${lighten(0.3, colore.actions[actioncolor])};
      color: ${lighten(0.1, colore.actions[actioncolor])};
    }
  `}
  ${({ admin }) => admin && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${colore.ui.neutralBg};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
  `}
  ${({ outline }) => outline && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${(props) => props.theme.brand};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
    svg {
      fill: ${(props) => props.fill ? props.fill : props.theme.brand};
    }
  `}
  ${({ bone }) => bone && css`
    color: ${colore.ui.contrast};
    border: 2px solid ${colore.ui.contrast};
    background-color: ${transparent};
    &:hover,
    &:active {
      background-color: rgba(0,0,0,.1);
    }
  `}
  &:active,
  &:focus {
    outline: none;
  }
  ${(props) => props.standard && css`
    min-width: 180px;
    max-width: 320px;
  `};
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${media.iphone45`
    ${({ standard }) => standard && 'min-width: 140px;'}
    padding: ${(props) => props.padding ? props.padding : '0.4em  0.6em;'};
    svg {
      margin: 0 6px 0 0;
    }
  `}
`;

export const Button = styled.button`
  ${ButtonStyle}
`;

Button.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const SubmitButton = styled.input`
  ${ButtonStyle}
`;

SubmitButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const ActionButtonStyle = css`
  font-size: 21px;
  padding: 21px 30px;
  background-color: ${(props) => props.bgcolor};
  border: ${(props) => props.border ? props.border : '2px solid white'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  font-weight: 600;
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${({ sectionbutton }) => sectionbutton && css`
    min-width: 50%;
    font-size: 18px;
    padding: 14px 30px;
    ${media.lt667`
      width: 100%;
    `}
  ${media.lt480`
    font-size: 16px;
    `}
  `}
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : colore.landing.green))};
    color: ${(props) => transparentize(0.5, (props.bgcolor ? props.bgcolor : colore.ui.contrast))};
  }
`;

export const ActionButton = styled(Button)`
  ${ActionButtonStyle};
`;

export const Icon = styled(Svg)`
  width: ${(props) => props.size ? props.size : '18px'};
  height: ${(props) => props.size ? props.size : '18px'};
  margin: ${(props) => props.margin};
  fill: ${(props) => props.fill ? props.fill : colore.ui.contrast};
  ${({ right }) => right && css`
    margin: 0 -6px 0 6px;
  `}
  ${({ left }) => left && css`
    margin: 0 6px 0 -6px;
  `}
  ${({ brandColor }) => brandColor && css`
    fill: ${(props) => props.theme.brand};
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

Icon.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const NoStyleButton = css`
  appearance: none;
  background: none;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: none;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const GhostButton = styled.button`
  ${NoStyleButton}
  display: inline-flex;
  align-items: center;
  justify-content: center;
  svg {
    fill: ${(props) => props.theme.brand};
  }
  ${(props) => props.fill && css`
    svg {
      fill: ${props.fill};
    }
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

GhostButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const TextButton = styled.button`
  color: ${(props) => props.color ? props.color : props.theme.brand};
  appearance: none;
  border: none;
  background: none;
  &:active,
  &:focus {
    outline: none;
  }
  &:hover {
    ${media.desktop`
      color: ${(props) => props.theme.darken};
    `}
  }
`;

TextButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    darken: '#1588AA',
  },
};
