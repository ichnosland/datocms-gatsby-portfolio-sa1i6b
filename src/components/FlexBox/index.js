/**
*
* FlexBox
*
*/
import styled, { css } from 'styled-components';

import Div from '../Div';
import media from 'style/mediainjector';

export const FlexBox = styled(Div)`
  display: flex;
  flex-direction: ${(props) => props.direction ? props.direction : 'row'};
  justify-content: ${(props) => props.justifyContent ? props.justifyContent : 'center'};
  align-items: ${(props) => props.alignItems ? props.alignItems : 'center'};
  align-content: ${(props) => props.alignContent ? props.alignContent : 'center'};
  flex-wrap: ${(props) => props.wrap ? props.wrap : 'nowrap'};
  ${media.print`
   ${({ noPrint }) => noPrint && css`
       display:none;
   `}
 `}
`;

export const FlexChild = styled(FlexBox)`
  order: ${(props) => props.order};
  flex-grow: ${(props) => props.flexGrow};
  flex-basis: ${(props) => props.flexBasis};
  align-self: ${(props) => props.alignSelf};
`;

export const FlexWrap = styled(FlexBox)`
  position: relative;
  flex-direction: column;
  justify-content: space-between;
  min-height: 100%;
  height: 100%;
  ${media.desktop`
    &:before {
      content: '';
      display: block;
    }
  `}
  ${media.mobile`
    display: block;
  `}
`;

export default FlexBox;
