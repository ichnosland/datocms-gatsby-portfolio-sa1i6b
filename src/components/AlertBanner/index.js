/**
*
* AlertBanner
*
*/

import styled, { css } from 'styled-components';
import { lighten, darken } from 'polished';
import FlexBox from '../FlexBox';
import media from 'style/mediainjector';
import { colore } from 'style/color';

export const BannerStyle = css`
  ${({ tag }) => tag && css`display: inline-flex;`};
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  text-align: center;
  margin: ${(props) => props.margin ? props.margin : '0'};
  border: ${(props) => props.border ? `1px solid ${props.theme.brand}` : `1px solid ${colore.ui.neutralBg}`};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  ${({ actioncolor }) => actioncolor && css`
    border: 1px solid ${colore.actions[actioncolor]};
    background-color: ${lighten(0.4, colore.actions[actioncolor])};
    color: ${darken(0.2, colore.actions[actioncolor])};
    svg {
      fill: ${darken(0.2, colore.actions[actioncolor])};
    }
  `}
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.general};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const AlertBanner = styled(FlexBox)`
  ${BannerStyle}
  padding: ${(props) => props.padding ? props.padding : '0.6em  1.6em'};
  svg {
    fill: ${({ color, actioncolor }) => {
    if (color) return color;
    if (actioncolor) return actioncolor;

    return colore.ui.txt;
  }};
  }
`;

AlertBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export const InfoBanner = styled(AlertBanner)`
  padding: ${(props) => props.padding ? props.padding : '0.6em  1em'};
  ${media.lt667`
    flex-direction: column;
    svg {
      margin: 10px 0;
    }
  `}
  ${media.iphone45`
    padding: ${(props) => props.padding ? props.padding : '0.6em'};
  `}
`;

InfoBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export const IconBanner = styled(FlexBox)`
  ${BannerStyle}
  align-self: stretch;
  align-items: center;
  padding: ${(props) => props.padding ? props.padding : '9px 16px'};
  div {
    margin-right: 12px;
    align-items: center;
    &:last-child {
      margin-right: 0;
    }
  }
  ${media.iphone45`
    padding: ${(props) => props.padding ? props.padding : '7px 12px'};
  `}
`;

IconBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export default AlertBanner;
