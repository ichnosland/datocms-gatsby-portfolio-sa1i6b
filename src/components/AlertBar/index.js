/**
*
* AlertBar
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import styled, { css } from 'styled-components';

import Div from '../Div';
import FlexBox from '../FlexBox';
import Svg from '../Svg';
import { IconBtn } from '../TopBar';
import { H3 } from '../Heading';
import media from 'style/mediainjector';
import { colore } from 'style/color';
import icon from 'icons/globals';
import { ContainerStyle } from '../Container';

export const AlertWrap = styled(Div)`
  position: absolute;
  top: 0;
  left: 0;
  color: ${colore.ui.contrast};
  width: 100%;
  height: 56px;
  padding: 0 20px;
  background-color: ${(props) => props.theme.brand};
  ${({ giusto }) => giusto && css`background-color: ${colore.actions.okay};`}
  ${({ sbagliato }) => sbagliato && css`background-color: ${colore.actions.error};`}
  ${({ aiuto }) => aiuto && css`background-color: ${colore.actions.help};`}
  z-index: 8;
  transition: all 0.5s;
  ${media.desktop`
    border-top-left-radius: ${(props) => props.versione ? '3px' : '20px'};
    border-top-right-radius: ${(props) => props.versione ? '3px' : '20px'};
  `}
  ${({ show }) => !show && css`
    display: none;
  `}
`;

AlertWrap.defaulProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const BarContainer = styled(FlexBox)`
  ${ContainerStyle}
  height: 100%;
  ${(props) => props.helpFx && css`
    justify-content: space-between;
    &::after {
      content: "";
      display: inline;
    }
  `}
  h3 {
    color: ${colore.ui.contrast};
  }
`;

class AlertBar extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      giusto,
      sbagliato,
      aiuto,
      versione,
      show,
      helpFx,
    } = this.props;
    return (
      <AlertWrap
        giusto={giusto}
        sbagliato={sbagliato}
        aiuto={aiuto}
        show={show}
        versione={versione}
      >
        <BarContainer helpFx={helpFx}>
          {helpFx &&
            <IconBtn onClick={helpFx}>
              <Svg {...icon.topBarFeedback} />
            </IconBtn>
          }
          <H3>
            {giusto &&
              'Risposta corretta!'
            }
            {sbagliato &&
              'Hai sbagliato!'
            }
            {aiuto &&
              'Aiuto'
            }
          </H3>
        </BarContainer>
      </AlertWrap>
    );
  }
}

AlertBar.propTypes = {
  giusto: PropTypes.bool,
  sbagliato: PropTypes.bool,
  aiuto: PropTypes.bool,
  versione: PropTypes.bool,
  show: PropTypes.bool.isRequired,
  helpFx: PropTypes.func,
};

export default AlertBar;
