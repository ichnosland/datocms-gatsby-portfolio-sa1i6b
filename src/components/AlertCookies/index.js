/**
*
* AlertCookies
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import Div from 'components/Div';
import P from 'components/Paragraph';
import { Button } from 'components/Button';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const AlertCookiesWrap = styled(Div)`
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 0 20px 20px;
  z-index: 1;
  ${media.lt667`
    padding: 0 10px 30px;
  `}
  ${({ show }) => !show && css`
    display: none;
  `}
`;

export const CookiesBanner = styled(Div)`
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  max-width: 860px;
  margin: 0 auto;
  font-size: 14px;
  border: ${(props) => props.border};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : 'rgba(255,255,255,.95)'};
  border-radius: 4px;
  border: 1px solid rgba(0,0,0,.15);
  box-shadow: 0 2px 1px rgba(0,0,0,.25);
  p {
    padding: 20px;
    margin: 0;
    &:last-child {
      padding: 0 20px 10px;
      text-align: right;
      ${media.lt480`
        padding: 0 20px 20px;
        text-align: center;
      `}
    }
  }
  ${media.lt667`
    font-size: 12px;
  `}
`;

class AlertCookies extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { show, color, border, bgcolor, onClickFunction } = this.props;
    return (
      <AlertCookiesWrap show={show}>
        <CookiesBanner
          color={color}
          border={border}
          bgcolor={bgcolor}
        >
          <P>Al fine di offrire il miglior servizio possibile questo sito utilizza cookies tecnici e di terze parti per funzionalità quali l&apos;accesso tramite social network e/o la visualizzazione di media. Se non acconsenti all&apos;utilizzo dei cookie di terze parti, alcune di queste funzionalità potrebbero essere non disponibili. Chiudendo questo banner o proseguendo con la navigazione acconsentirai al loro impiego. Per saperne di pi&ugrave; o negare il consenso a tutti o ad alcuni cookie puoi consultare <a href="https://maieuticallabs.it/privacy" target="_blank"><span>l&apos;informativa sulla privacy</span></a>.</P>
          <P><Button actioncolor="hint" standard={1} onClick={onClickFunction}>Chiudi</Button></P>
        </CookiesBanner>
      </AlertCookiesWrap>
    );
  }
}

AlertCookies.propTypes = {
  color: PropTypes.string,
  border: PropTypes.string,
  bgcolor: PropTypes.string,
  show: PropTypes.bool.isRequired,
  onClickFunction: PropTypes.func,
};

export default AlertCookies;
