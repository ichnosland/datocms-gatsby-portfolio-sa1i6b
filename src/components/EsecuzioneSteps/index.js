/**
*
* EsecuzioneSteps
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import Container from 'components/Container';
import CentralBox from 'components/CentralBox';
import ProgressBar from 'components/ProgressBar';
import TopBar from 'components/TopBar';
import AlertBar from 'components/AlertBar';
import ToolBar from 'components/ToolBar';
import ToolBarButton from 'components/ToolBar/ToolBarButton';
import Step from './Step';

class EsecuzioneSteps extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  constructor(props) {
    super(props);
    this.toolBarButtonInvio = React.createRef();
  }

  componentDidMount() {
    this.checkFocusButtonRisposta();
  }

  componentDidUpdate() {
    this.checkFocusButtonRisposta();
  }

  checkFocusButtonRisposta = () => {
    const {
      correzioneRisposta: { isCheckable, mostraSoluzione, mostraCorrezione },
      stepCaricato = { esercizi: [] },
    } = this.props;
    const isScrivi = ['U', 'P'].indexOf((stepCaricato.esercizi[0] || {}).tipo) > -1;

    if ((isCheckable && !isScrivi) || mostraSoluzione || mostraCorrezione) {
      this.enableFocusButtonRisposta();
    }
  }

  enableFocusButtonRisposta = /* istanbul ignore next */ () => {
    const { disableAutofocus } = this.props;

    if (disableAutofocus || !((this.toolBarButtonInvio || {}).current || {}).focus) {
      return false;
    }

    this.toolBarButtonInvio.current.focus();
    return true;
  }

  calcolaPercentuale = () => {
    const { stepTotali, stepEseguiti } = this.props;
    if (stepTotali > 0 && stepEseguiti > -1) {
      return Math.round((stepEseguiti / stepTotali) * 100);
    }
    return -1;
  }
  render() {
    const {
      stepCaricato,
      topBarParams,
      hasLessico,
      funzioneSelezionaRisposta,
      rispostaSelezionata,
      apriModalBox,
      chiudiModalBox,
      hasHelp,
      hasSkip,
      correzioneRisposta,
      correzioneRisposta: {
        isChecked,
        isCorrect,
        isHelpEnabled,
        mostraSoluzione,
        isCheckable,
        mostraCorrezione,
        isFocusEnabled,
      },
      correzioneRispostaFx,
      helpFunction,
      skipFunction,
      hasCorreggi,
      helpButtonFunction,
      disableAutofocus,
      toggleFocusFunction,
      isVersione,
      audioSettings,
    } = this.props;

    const percentualeSvolgimento = this.calcolaPercentuale();
    let titleButton = 'Continua';
    if (hasCorreggi) {
      if (isChecked && !isCorrect && !isHelpEnabled) {
        titleButton = 'Riprova';
      }
    } else if (!isChecked) {
      titleButton = 'Controlla';
    }

    const toolbarDefaultButtonInvia = {
      enabled: isCheckable || mostraSoluzione,
      full: !hasSkip && (!hasHelp || isChecked),
      check: hasSkip || (hasHelp && !isChecked),
      title: titleButton,
      onClickFx: isCheckable || mostraSoluzione ?
        correzioneRispostaFx : null,
      refData: this.toolBarButtonInvio,
    };

    return (
      <Container>
        <CentralBox
          versione={isVersione}
        >
          {percentualeSvolgimento > -1 &&
            <ProgressBar 
              lezione={!isVersione}
              versione={isVersione}
              percentuale={percentualeSvolgimento}
            />
          }
          <TopBar
            box
            esecuzione
            noShadow
            {...topBarParams}
          />
          <AlertBar
            versione={isVersione}
            giusto={!isHelpEnabled && isCorrect}
            sbagliato={!isHelpEnabled && !isCorrect}
            aiuto={isHelpEnabled}
            show={mostraCorrezione || mostraSoluzione}
            helpFx={helpButtonFunction}
          />
          <Step
            step={stepCaricato}
            hasLessico={hasLessico}
            funzioneSelezionaRisposta={funzioneSelezionaRisposta}
            rispostaSelezionata={rispostaSelezionata}
            apriModalBox={apriModalBox}
            chiudiModalBox={chiudiModalBox}
            hasHelp={hasHelp}
            hasSkip={hasSkip}
            hasCorreggi={hasCorreggi}
            correzioneRisposta={correzioneRisposta}
            correzioneRispostaFx={correzioneRispostaFx}
            helpFunction={helpFunction}
            skipFunction={skipFunction}
            disableAutofocus={disableAutofocus}
            enableFocusButtonRisposta={this.enableFocusButtonRisposta}
            toggleFocusFunction={toggleFocusFunction}
            audioSettings={audioSettings}
          />
          <ToolBar hideOnFocus={!(mostraCorrezione || mostraSoluzione) && isFocusEnabled}>
            {hasHelp && !isChecked && <ToolBarButton
              help
              onClickFx={helpFunction}
            />}
            {hasSkip && <ToolBarButton
              skip
              onClickFx={skipFunction}
            />}
            <ToolBarButton {...toolbarDefaultButtonInvia} />
          </ToolBar>
        </CentralBox>
      </Container>
    );
  }
}

EsecuzioneSteps.propTypes = {
  stepCaricato: PropTypes.shape({
    esercizi: PropTypes.array.isRequired,
    testi: PropTypes.array.isRequired,
  }),
  riaccodaSbagliate: PropTypes.func,
  rieseguiSbagliate: PropTypes.func,
  mostraCorrezione: PropTypes.func,
  funzioneSalta: PropTypes.func,
  funzioneEsegui: PropTypes.func,
  funzioneSelezionaRisposta: PropTypes.func,
  stepTotali: PropTypes.number,
  stepEseguiti: PropTypes.number,
  hasLessico: PropTypes.bool,
  isVersione: PropTypes.bool,
  rispostaSelezionata: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
  ]),
  hasHelp: PropTypes.bool.isRequired,
  hasSkip: PropTypes.bool,
  hasCorreggi: PropTypes.bool.isRequired,
  apriModalBox: PropTypes.func.isRequired,
  chiudiModalBox: PropTypes.func.isRequired,
  correzioneRisposta: PropTypes.shape({
    isCheckable: PropTypes.bool.isRequired,
    isChecked: PropTypes.bool.isRequired,
    isPristine: PropTypes.bool.isRequired,
    isCorrect: PropTypes.bool.isRequired,
    isHelpEnabled: PropTypes.bool.isRequired,
    mostraSoluzione: PropTypes.bool.isRequired,
    mostraCorrezione: PropTypes.bool.isRequired,
  }).isRequired,
  correzioneRispostaFx: PropTypes.func.isRequired,
  helpFunction: PropTypes.func,
  skipFunction: PropTypes.func,
  topBarParams: PropTypes.object.isRequired,
  helpButtonFunction: PropTypes.func,
  disableAutofocus: PropTypes.bool,
  toggleFocusFunction: PropTypes.func.isRequired,
  audioSettings: PropTypes.shape({
    maxRepetitions: PropTypes.number,
    enablePause: PropTypes.bool,
  }),
};

export default EsecuzioneSteps;
