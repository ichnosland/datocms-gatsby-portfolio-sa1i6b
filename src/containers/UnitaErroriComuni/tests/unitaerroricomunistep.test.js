import React from 'react';
import { shallow } from 'enzyme';

import { mockElementiN } from 'common/testing-mocks';
import EsecuzioneSteps from 'components/EsecuzioneSteps';
import { FlexWrap } from 'components/FlexBox';

import UnitaErroriComuniStep from '../UnitaErroriComuniStep';


const mockConfiguration = {
  product: 'product',
  blocco: true,
  disciplinaId: 123,
  homePage: '/homepage',
};

const mockProps = {
  totaleDomande: 3,
  stepEseguiti: 2,
  backUrl: '/back-url',
  risposta: {
    rispostaSelezionata: {},
    isCheckable: true,
    isChecked: false,
    isPristine: true,
    isCorrect: true,
    isInterfaceLocked: false,
    isHelpEnabled: true,
    correzioneStep: {
      corrette: [],
      sbagliate: [],
    },
    mostraSoluzione: false,
    mostraCorrezione: true,
    isFocusEnabled: false,
    risposteInizializza: false,
  },
  step: {
    esercizi: [mockElementiN[1]],
    testi: [mockElementiN[0]],
    numeroProgressivoStep: 0,
    isStepLoaded: true,
  },
  correzioneRispostaFx: () => { },
  apriModalBoxFx: () => { },
  chiudiModalBoxFx: () => { },
  helpFunction: () => { },
  selezioneRispostaFx: () => { },
  toggleFocusFunction: () => { },
  configuration: mockConfiguration,
  apriChiudiSezioniFx: () => { },
  helpButtonFunction: () => { },
};

describe('<UnitaErroriComuniStep />', () => {
  it('visualizza dati', () => {
    const renderedComponent = shallow(
      <UnitaErroriComuniStep {...mockProps} />
    );
    expect(renderedComponent.find(FlexWrap).length).toBe(1);
    expect(renderedComponent.find(EsecuzioneSteps).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });
});
