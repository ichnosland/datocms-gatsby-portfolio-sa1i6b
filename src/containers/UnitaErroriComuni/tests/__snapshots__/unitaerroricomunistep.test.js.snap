// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnitaErroriComuniStep /> visualizza dati 1`] = `
<FlexBox__FlexWrap>
  <EsecuzioneSteps
    apriModalBox={[Function]}
    chiudiModalBox={[Function]}
    correzioneRisposta={
      Object {
        "correzioneStep": Object {
          "corrette": Array [],
          "sbagliate": Array [],
        },
        "isCheckable": true,
        "isChecked": false,
        "isCorrect": true,
        "isFocusEnabled": false,
        "isHelpEnabled": true,
        "isInterfaceLocked": false,
        "isPristine": true,
        "mostraCorrezione": true,
        "mostraSoluzione": false,
        "risposteInizializza": false,
      }
    }
    correzioneRispostaFx={[Function]}
    funzioneSelezionaRisposta={[Function]}
    hasCorreggi={false}
    hasHelp={true}
    hasLessico={true}
    helpButtonFunction={[Function]}
    helpFunction={[Function]}
    key="esecuzione_preview_0"
    rispostaSelezionata={Object {}}
    stepCaricato={
      Object {
        "esercizi": Array [
          Object {
            "continua": false,
            "distrattore_1": "passivo",
            "distrattore_2": "imperfetto",
            "distrattore_3": "",
            "distrattore_4": "",
            "esercizioId": 97552,
            "etichetta_1": null,
            "etichetta_2": null,
            "etichetta_3": null,
            "etichetta_4": null,
            "etichetta_5": null,
            "focus": false,
            "form": Array [
              "imperfetto",
              "passivo",
              "attivo",
              "presente",
            ],
            "html": null,
            "id": 94635,
            "ignora_lessico": false,
            "inputLessicale": Array [],
            "italiano": false,
            "parole_gia_ricostruite": "",
            "risposta_1": "attivo",
            "risposta_2": "",
            "risposta_3": "",
            "risposta_4": "",
            "testo_principale": "presente",
            "testo_principale_post": "",
            "testo_principale_pre": "",
            "tipo": "N",
            "titolo": "Analizza il tempo e la diatesi",
            "traduci": Array [],
          },
        ],
        "isStepLoaded": true,
        "numeroProgressivoStep": 0,
        "testi": Array [
          Object {
            "continua": false,
            "distrattore_1": "",
            "distrattore_2": "",
            "distrattore_3": "",
            "distrattore_4": "",
            "esercizioId": 97552,
            "etichetta_1": null,
            "etichetta_2": null,
            "etichetta_3": null,
            "etichetta_4": null,
            "etichetta_5": null,
            "focus": true,
            "form": null,
            "html": "<span lessico=\\"Luna\\" idx=\\"0\\" testo=\\"testo\\" show-modal=\\"setModal()\\" tables=\\"updateTables(input)\\">Luna</span> <span class=\\"contesto\\"><span lessico=\\"splendet\\" idx=\\"1\\" testo=\\"testo\\" show-modal=\\"setModal()\\" tables=\\"updateTables(input)\\">splendet</span> </span><span lessico=\\"\\" idx=\\"2\\" testo=\\"testo\\" show-modal=\\"setModal()\\" tables=\\"updateTables(input)\\"></span>, <span lessico=\\"stellae\\" idx=\\"3\\" testo=\\"testo\\" show-modal=\\"setModal()\\" tables=\\"updateTables(input)\\">stellae</span> <span lessico=\\"fulgent\\" idx=\\"4\\" testo=\\"testo\\" show-modal=\\"setModal()\\" tables=\\"updateTables(input)\\">fulgent</span>.",
            "id": 94634,
            "ignora_lessico": false,
            "inputLessicale": Array [
              Object {
                "focus": false,
                "forma": "luna",
                "lemma": "luna",
                "tabelle": null,
                "tipo": "n",
                "traduzione": "luna",
              },
              Object {
                "focus": false,
                "forma": "splendet",
                "lemma": "splendeo",
                "tabelle": null,
                "tipo": "v",
                "traduzione": "splendere",
              },
              Object {
                "focus": false,
                "forma": "stellae",
                "lemma": "stella",
                "tabelle": null,
                "tipo": "n",
                "traduzione": "stella",
              },
              Object {
                "focus": false,
                "forma": "fulgent",
                "lemma": "fulgeo",
                "tabelle": null,
                "tipo": "v",
                "traduzione": "brillare",
              },
            ],
            "italiano": false,
            "parole_gia_ricostruite": "",
            "risposta_1": "",
            "risposta_2": "",
            "risposta_3": "",
            "risposta_4": "",
            "testo_principale": "Luna <span class=\\"contesto\\">splendet</span>, stellae fulgent.",
            "testo_principale_post": "",
            "testo_principale_pre": "",
            "tipo": false,
            "titolo": null,
            "traduci": Array [],
          },
        ],
      }
    }
    stepEseguiti={2}
    stepTotali={3}
    toggleFocusFunction={[Function]}
    topBarParams={
      Object {
        "closeBtn": Object {
          "enabled": true,
          "url": "/back-url",
        },
      }
    }
  />
  <LogoPlatform
    desktopOnly={true}
    execLogo={true}
    margin="0 0 20px"
    product="product"
  />
</FlexBox__FlexWrap>
`;
